### ISSUE TITLE
Helmfile incompatible with kustomize 4.0.1

### ISSUE THREAD

**Issue by EricPE69** [roboll/helmfile#1688]
With kustomize 4.0.1 helmfile does not work:
`
COMMAND:
  kustomize -o /tmp/chartify266988612/mongodb-574f96b8db/templates/kustomized.yaml build --load_restrictor=none --enable_alpha_plugins /tmp/chartify266988612/mongodb-574f96b8db

OUTPUT:
  Error: unknown flag: --load_restrictor]
`
There is a breaking change in kustomize 4.0.1:
see https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.0.1

`In flags, convert underscores to hyphens to be compatible with kubectl.

These v4 flags are now incompatible with v3.`

Value for --load-restrictor is also invalid: must be LoadRestrictionsNone and not none.

**Comment by mwakkach** [comment]
same issue here !!!

**Comment by strainovic** [comment]
Same here :(

**Comment by mumoshu** [comment]
Ah, this is too bad! We had no way to detect kustomize version before run so far so we'd need to build that to go forward.

**Comment by mumoshu** [comment]
Do we really need to parse this to detect kustomize version? Can it be reliable? 🤦 

```
$ kustomize version --short
{kustomize/v3.8.1  2020-07-16T00:58:46Z  }
```

**Comment by mumoshu** [comment]
Please see https://github.com/variantdev/chartify/commit/cfeb7ff5ffd17e714cc0603085e8bed85d4ab33b

**Comment by mumoshu** [comment]
The fix for this is #1758
